pkg/util/runtime/runtime.go

```diff
@@ -110,3 +113,24 @@ func NewServerMux() *gwruntime.ServeMux {
 	)
 	return mux
 }
+
+// InClusterBuildConfig is a helper function that first attempts to build configurations
+// using InClusterConfig(). If InClusterConfig is unsuccessful, it then tries to build
+// configurations from a kubeconfigPath. This path is typically passed in as a command line
+// flag for cluster components. If neither the InClusterConfig nor the kubeconfigPath
+// are successful, the function logs a warning and falls back to a default configuration.
+func InClusterBuildConfig(kubeconfigPath string) (*restclient.Config, error) {
+	kubeconfig, err := restclient.InClusterConfig()
+	if err == nil {
+		return kubeconfig, nil
+	}
+	klog.Warning("error creating inClusterConfig, trying to build config from flags", err)
```

Comment:
Oooh! Good thing to fix I just noticed.

Rather than using klog, let's stick with our usual logger library (logrus), so we can be consistent (see suggestion above), and pass in the `logger` from the `main.go` where this function gets called.

Reply:
I've made the changes as per your suggestion.👍